### PR TITLE
Vendor gardener/gardener@v1.53.0

### PR DESCRIPTION
**Pull request by tedteng**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement
/platform alicloud

**What this PR does / why we need it**:
Vendor gardener/gardener@v1.52.1 to be in sync with the other provider extensions for use Bastion config validator feature
https://github.com/gardener/gardener/releases/tag/v1.52.1

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency is updated:
- github.com/gardener/gardener: v1.50.1 -> v1.53.0
```

/invite @dkistner @shafeeqes Thanks

### CODE REVIEW

File: pkg/controller/bastion/actuator_reconcile.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (be *bastionEndpoints) Ready() bool {
 	return be != nil && IngressReady(be.private) && IngressReady(be.public)
 }
 
-func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Bastion, cluster *controller.Cluster) error {
+func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, bastion *extensionsv1alpha1.Bastion, cluster *controller.Cluster) error {
 	logger := a.logger.WithValues("bastion", client.ObjectKeyFromObject(bastion), "operation", "reconcile")
```
Comment (shafeeqes) [inline pkg/controller/bastion/actuator_reconcile.go:54]: unlike `calico` extension, here logger is used in some functions, please adapt accordingly.
Ref: https://github.com/gardener/gardener-extension-provider-aws/pull/585/commits/78f27287765291e1b6c3481774c5e5e7040f5d8f

File: test/integration/bastion/bastion_test.go [hunk 2]
```diff
@@ -126,11 +125,7 @@ var _ = BeforeSuite(func() {
 	alicloud.InternalChartsPath = filepath.Join(repoRoot, alicloud.InternalChartsPath)
 
 	// enable manager logs
-	logf.SetLogger(zap.New(zap.UseDevMode(true), zap.WriteTo(GinkgoWriter)))
-
-	log := logrus.New()
-	log.SetOutput(GinkgoWriter)
-	logger = logrus.NewEntry(log)
+	logger = log.Log.WithName("test")
```
Comment (shafeeqes) [inline test/integration/bastion/bastion_test.go:128]: Please adapt similar to https://github.com/gardener/gardener-extension-provider-aws/pull/585/commits/b01bfb96b3c8fb3e0267a1715cd4171a3016ac32

File: Makefile [hunk 3]
```diff
@@ -98,14 +98,6 @@ docker-images:
 # Rules for verification, formatting, linting, testing and cleaning #
 #####################################################################
 
-.PHONY: install-requirements
```
Comment (shafeeqes) [inline Makefile:101]: Can you rebase your branch?
This https://github.com/gardener/gardener-extension-provider-alicloud/pull/516 is already merged.

File: pkg/controller/bastion/actuator_reconcile.go (hunk 1 above)
Comment (tedteng) [reply pkg/controller/bastion/actuator_reconcile.go]: PR feedback d5aa79f6

File: Makefile (hunk 3 above)
Comment (tedteng) [reply Makefile]: done

File: test/integration/bastion/bastion_test.go (hunk 2 above)
Comment (tedteng) [reply test/integration/bastion/bastion_test.go]: a42163db done

Comment (shafeeqes) [reply test/integration/bastion/bastion_test.go]: You can cleanup the `logger` from actuator structs, wherever possible. ref: https://github.com/gardener/gardener/pull/6332/commits/1a3570f086eacf77daa2c297d9296bad86dfef0a

**Comment by shafeeqes** [comment]
Also, you could adapt https://github.com/gardener/gardener-extension-networking-calico/pull/201/commits/31c7444cb748f91ebcb8cbaf4317e0a377087ac2
if vendoring runs fine without this replace directive.

**Review by dkistner — APPROVED** [review]
/lgtm
/squash

File: test/integration/bastion/bastion_test.go (hunk 2 above)
Comment (tedteng) [reply test/integration/bastion/bastion_test.go]: 5ce0129e  cleanup `logger` from bastion actuator structs.  let me know if correct, besides, should I create one separate ticket for cleanup `logger` for other objects under the controller folder?

**Comment by tedteng** [comment]
> Also, you could adapt [gardener/gardener-extension-networking-calico@31c7444](https://github.com/gardener/gardener-extension-networking-calico/commit/31c7444cb748f91ebcb8cbaf4317e0a377087ac2) if vendoring runs fine without this replace directive.

done 8e92bb9c

File: test/integration/bastion/bastion_test.go (hunk 2 above)
Comment (tedteng) [reply test/integration/bastion/bastion_test.go]: 4259a5bae9679cbb43aed45ac961dceeaefde2fc
495d17b692f2c508b0c09b24fcd628a1eb41f544

**Comment by shafeeqes** [comment]
> > Also, you could adapt [gardener/gardener-extension-networking-calico@31c7444](https://github.com/gardener/gardener-extension-networking-calico/commit/31c7444cb748f91ebcb8cbaf4317e0a377087ac2) if vendoring runs fine without this replace directive.
> 
> ~done [8e92bb9](https://github.com/gardener/gardener-extension-provider-alicloud/commit/8e92bb9caf39aceaf29bba8457379dfc60fbc273)~
> 
> update: an error after `make revendor` https://concourse.ci.gardener.cloud/builds/5539376. I have no idea about that then I revert it. for the rest of things, all done now.
> 
> ```
> github.com/gardener/gardener-extension-provider-alicloud/pkg/controller/worker imports
> 	github.com/gardener/gardener/pkg/client/kubernetes imports
> 	github.com/gardener/hvpa-controller/api/v1alpha1 loaded from github.com/gardener/hvpa-controller/api@v0.5.0,
> 	but go 1.16 would fail to locate it:
> 	ambiguous import: found package github.com/gardener/hvpa-controller/api/v1alpha1 in multiple modules:
> 	github.com/gardener/hvpa-controller v0.3.1 (/go/pkg/mod/github.com/gardener/hvpa-controller@v0.3.1/api/v1alpha1)
> 	github.com/gardener/hvpa-controller/api v0.5.0 (/go/pkg/mod/github.com/gardener/hvpa-controller/api@v0.5.0/v1alpha1)
> 
> To proceed despite packages unresolved in go 1.16:
> 	go mod tidy -e
> If reproducibility with go 1.16 is not needed:
> 	go mod tidy -compat=1.17
> For other options, see:
> 	https://golang.org/doc/modules/pruning
> make[1]: *** [/go/src/github.com/gardener/gardener-extension-provider-alicloud/Makefile:103: revendor] Error 1
> make[1]: Leaving directory '/go/src/github.com/gardener/gardener-extension-provider-alicloud'
> make: *** [Makefile:117: check-generate] Error 1
> ```

Ref: https://github.com/gardener/hvpa-controller/issues/92

seems like this depends on the go version. this extension is using an older version.

**Review by shaoyongfeng — APPROVED** [review]

**Pull request merged**